src/libcore/num/mod.rs

```diff
@@ -745,7 +760,7 @@ $EndFeature, "
             #[inline]
             pub const fn checked_add(self, rhs: Self) -> Option<Self> {
                 let (a, b) = self.overflowing_add(rhs);
-                if b {None} else {Some(a)}
+                if unlikely!(b) {None} else {Some(a)}
```

Comment:
Will it have the same effect by flipping the case?
```rust
if !b { Some(a) } else { None }
```

Reply:
Rearranging the code gives no hint to LLVM that one branch has more weight than another.

Reply:
I tried it. The code is better for only this method alone, not at callsite. But at callsite, things are not improved.